src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/NuGetProjectUpgradeWindow.xaml

```diff
@@ -0,0 +1,305 @@
+<nuget:VsDialogWindow
+  x:Class="NuGet.PackageManagement.UI.NuGetProjectUpgradeWindow"
+  xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
+  xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+  xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
+  xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
+  xmlns:nuget="clr-namespace:NuGet.PackageManagement.UI"
+  mc:Ignorable="d"
+  x:Name="_self"
+  Background="{DynamicResource {x:Static nuget:Brushes.BackgroundBrushKey}}"
+  Foreground="{DynamicResource {x:Static nuget:Brushes.UIText}}"
+  ShowInTaskbar="False"
+  WindowStartupLocation="CenterOwner"
+  Title="{x:Static nuget:Resources.WindowTitle_NuGetUpgrader}"
+  Height="600"
+  MinHeight="480"
+  MinWidth="640"
+  Width="640"
+  d:DataContext="{d:DesignInstance Type=nuget:NuGetProjectUpgradeWindowModel, IsDesignTimeCreatable=True}" HasDialogFrame="False" ResizeMode="CanResizeWithGrip">
+
+  <Window.Resources>
+    <ResourceDictionary>
+      <ResourceDictionary.MergedDictionaries>
+        <nuget:SharedResources />
+      </ResourceDictionary.MergedDictionaries>
+      <BitmapImage
```

Comment:
use the VS image service instead of `BitmapImage`. [Here](https://github.com/NuGet/NuGet.Client/commit/67a9733f03060e66892967c2d7004d4cc8ea66c2#diff-d927676440988ef9c04746e3b5131aa1R114) is an example of how to use it.